tests/neg/i1679.scala

```diff
@@ -1,5 +1,5 @@
 class A[T]
 object o {
   // Testing compiler crash, this test should be modified when named type argument are completely implemented
-  val x: A[T=Int, T=Int] = ??? // error: ']' expected, but '=' found
+  val x: A[T=Int, T=Int] = ??? // error: ']' expected, but '=' found // error
```

Comment:
Also here, there seems to be no message accompanying the message

Reply:
The second message here is not ideal: 
```
Declaration of value x not allowed here: only classes can have declared but undefined members```. 
This is arguably a regression. I'm not quite sure what's happening, but I noticed there were a bunch of untagged, less-than-ideal error messages elsewhere so I left it. Think it's worth fixing here? Leaving a TODO?

Reply:
Not sure, if there are other wierd errors then I guess we can create a Todo with an issue to improve this.

Reply:
I looked into it a bit more. When parsing a comma-separated list, if thing you're looking for (in this case, a type tree) ends before you hit a comma, the parser thinks you're done the list, so the parser reads `T` then thinks it's done the list because it doesn't see a `,`. It's then expecting to see a `]`, and issues the first error. Then it skips to the next safe point, which in the past was the next `]`, but now is the `,`. So I guess the parser sees this as `val x: A[T], T = Int] = ???`, so that explains why the parser produces a `val x: A[T]` and hence the new error. I don't know why there aren't errors resulting from the `, T = Int] = ???` suffix, but presumably there are and they are just suppressed?

Reply:
I pushed a change that gets rid of this (bad) error by changing `skip` to look for a comma only if we're not explicitly trying to find a statement separator.

Reply:
Should we remove the added error then?